core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -153,7 +153,12 @@ class GroupCoordinator(val brokerId: Int,
             // coordinator OR the group is in a transient unstable phase. Let the member retry
             // joining without the specified member id,
             responseCallback(joinError(memberId, Errors.UNKNOWN_MEMBER_ID))
-
+          case InitialRebalance =>
+            if (memberId != JoinGroupRequest.UNKNOWN_MEMBER_ID) {
+              responseCallback(joinError(memberId, Errors.UNKNOWN_MEMBER_ID))
```

Comment:
Furthermore, I feel we can merge this state with the `prepareRebalance`, see my comments above.